### ISSUE TITLE
"Error calling sync triggers (InternalServerError)" - when pushing with cli

### ISSUE THREAD

**Issue by forki** [Azure/azure-functions-cli#315]
![image](https://user-images.githubusercontent.com/57396/32771188-3af159b4-c922-11e7-82b1-4b6843f9d4c8.png)

what can I do?

**Comment by davidebbo** [comment]
Possibly the same as https://github.com/projectkudu/kudu/issues/2622 (which is not yet understood).

**Comment by forki** [comment]
I'm getting it now for every single deploy

**Comment by davidebbo** [comment]
We're still investigating this, but does restarting the App help? Is this using  Consumption plan or App Service Plan?

**Comment by forki** [comment]
No restart did not help. It's even showing weird error messages on stop and
start. IIRC we are on app service plan here.

Am 14.11.2017 16:26 schrieb "David Ebbo" <notifications@github.com>:

**Comment by davidebbo** [comment]
Can you share your function app name (that's the **app** name, not function name!), either directly or [indirectly](https://github.com/Azure/azure-webjobs-sdk-script/wiki/Sharing-Your-Function-App-name-privately)? This will help us investigate. Thanks!

**Comment by forki** [comment]
Sure:

    2017-11-14T15:34:05.736 Function started (Id=a54c4bee-44af-467d-935c-2e6b8d23585f)

Region is West Europe and it's WestEuropePlan (Consumption)

**Comment by davidebbo** [comment]
Thanks, we will try to investigate later today and get back to you.

**Comment by forki** [comment]
thank you so much!

**Comment by julioct** [comment]
I'm getting the exact same issue here. Function app name is **EpsFunctions-Dev**

**Comment by davidebbo** [comment]
It's a bug and we're looking into it now. Also came up on https://stackoverflow.com/questions/47294382/azure-functions-continuous-deployment-failing-on-syncing-function-triggers-step

**Comment by davidebbo** [comment]
Issue should now be resolved. Please verify.

**Comment by forki** [comment]
I can confirm, thanks for fast fix!

**Comment by Tomaszenko** [comment]
I am getting exactly the same error with following tools versions:
Azure Functions Core Tools (2.0.1-beta.22)
Function Runtime Version: 2.0.11415.0

Should I use Azure Functions Core Tools 1.x to avoid it?

**Comment by davidebbo** [comment]
@Tomaszenko can you [share the app name](https://github.com/Azure/azure-functions-host/wiki/Sharing-Your-Function-App-name-privately) and the UTC time of the failed attempt?

**Comment by kierenh** [comment]
@davidebbo Also seeing this:
`~\SBY.kwidgebo.FunctionsApp>func azure functionapp publish SBY-kwidgebo --publish-local-settings -i --overwrite-settings -y
Getting site publishing info...
Publish ~\SBY.kwidgebo.FunctionsApp contents to an Azure Function App. Locally deleted files are not removed from destination.
Creating archive for current directory...
Uploading archive...
Error calling sync triggers (InternalServerError).
Retry: 1 of 2
Creating archive for current directory...
Uploading archive...
Error calling sync triggers (InternalServerError).
Retry: 2 of 2
Creating archive for current directory...
Uploading archive...
Error calling sync triggers (InternalServerError).`

**Comment by davidebbo** [comment]
@kierenh please share additional info as per my previous reply.

**Comment by kierenh** [comment]
Approx 12:40 - to 12:50 am, Monday, 30 April 2018

**Comment by davidebbo** [comment]
- please confirm that this is UTC
- provide app name data per that same answer above

**Comment by kierenh** [comment]
When I tried to run the function from Azure I got his error:
`Error:

The function runtime is unable to start. Microsoft.Azure.WebJobs.Script: Unable to parse host.json file. Newtonsoft.Json: After parsing a value an unexpected character was encountered: ". Path 'functions[0]', line 4, position 4.
Session Id: 2dc89758d9c1494aa06b4fc9252a2d04

Timestamp: 2018-04-30T01:16:54.862Z`

Logical - there was a syntax error in the host.json file of the Function App I was publishing.

Once the syntax error was resolved, I was able to run the function. The function name btw is:
`2018-04-30T01:20:06  Welcome, you are now connected to log-streaming service.
2018-04-30T01:20:40.204 [Info] Function started (Id=f322a377-b3b2-4b2e-ab71-a35e4110f880)
2018-04-30T01:20:40.204 [Info] C# HTTP trigger function processed a request.
2018-04-30T01:20:40.204 [Info] Function completed (Success, Id=f322a377-b3b2-4b2e-ab71-a35e4110f880, Duration=12ms)`

And with the Function running, I tried publishing via Azure Functions CLI and it was successful so I suspect the root cause of my issue was because the Azure Function runtime wasn't able to start in the first place.
